go/vt/vtgate/engine/insert_select.go

```diff
@@ -111,8 +109,11 @@ func (ins *InsertSelect) TryStreamExecute(ctx context.Context, vcursor VCursor,
 		}
 		return callback(res)
 	}
-	ctx, cancelFunc := addQueryTimeout(ctx, vcursor, ins.QueryTimeout)
-	defer cancelFunc()
+	if ins.QueryTimeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(ins.QueryTimeout)*time.Millisecond)
```

Comment:
Would it make sense to rename `QueryTimeout` to `QueryTimeoutMillis`? I think it would improve readability of the code.

Reply:
I would leave it as-is for now, as it is there in few other primitives as well.